--- a/src/Makefile_Evas.am	2014-12-09 11:11:10.675614077 +0100
+++ b/src/Makefile_Evas.am	2014-12-09 11:13:45.383609496 +0100
@@ -866,6 +866,7 @@
 engineglgenericpkg_LTLIBRARIES = modules/evas/engines/gl_generic/module.la
 modules_evas_engines_gl_generic_module_la_SOURCES = $(GL_GENERIC_SOURCES)
 modules_evas_engines_gl_generic_module_la_CFLAGS = \
+-DLINUX -DEGL_API_FB \
 -I$(top_builddir)/src/lib/efl \
 -I$(top_srcdir)/src/lib/evas/include \
 -I$(top_srcdir)/src/lib/evas/cserve2 \
